api/src/controllers/forms.js

```diff
@@ -27,11 +28,15 @@ const isCollectForm = doc => doc && doc.context && doc.context.collect;
 // @param {Object} data - couchdb view data
 // @param {Object} req - the request object
 const listFormsXML = (forms, req) => {
-  const urls = forms
+  const downloadUrl = `${req.protocol}://${req.headers.host}/api/v1/forms/` + '${formId}.xml'; // formId is replaced by openrosaFormList
+  const xmls = forms
     .filter(form => isCollectForm(form))
-    .map(form => `${req.protocol}://${req.headers.host}/api/v1/forms/${form.internalId}.xml`);
+    .map(form => formsService.getXFormAttachment(form))
+    .filter(attachment => !!attachment)
+    .map(attachment => Readable.from(attachment.data));
+
   return new Promise((resolve, reject) => {
-    openrosaFormList(urls, (err, xml) => {
+    openrosaFormList(xmls, { downloadUrl }, (err, xml) => {
```

Comment:
Passing the xmls means openrosaFormList doesn't have to request them which is a problem now that it's inside cht-net.